docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,116 @@
+# ScheduledJob Controller
+
+## Abstract
+A proposal for implementing a new controller - ScheduledJob controller, which
+will be responsible for managing time based jobs, namely:
+* once at a specified point in time,
+* repeatedly at a specified point in time.
+
+There is already an upstream discussion regarding that particular subject:
+* Distributed CRON jobs [#2156](https://github.com/GoogleCloudPlatform/kubernetes/issues/2156)
+
+There are also similar solutions available already:
+* [Mesos Chronos](https://github.com/mesos/chronos)
+* [Quartz](http://quartz-scheduler.org/)
+
+
+## Use Cases
+1. Be able to schedule a job execution at a given point in time.
+1. Be able to create a repetitive job, eg. database backup, sending emails.
+1. Be able to create a chain of jobs dependent one on another.
+1. Be able to specify triggers on job’s success/failure.
+
+
+## Motivation
+ScheduledJobs are needed for performing all time related actions, namely backups,
+report generation and alike.  Each of these tasks should be allowed to perform
+repeatedly (once a day/month, etc.) or once at a given point in time.
+
+
+## Implementation
+ScheduledJob controller relies heavily on the [Job Controller API](https://github.com/openshift/origin/blob/master/docs/proposals/job.md)
+for running actual jobs, on top of which it adds information regarding the date
+and time part according to ISO8601 format.
+
+The new `ScheduledJob` object will have the following content:
+
+```go
+// ScheduledJob represents the configuration of a single scheduled job.
+type ScheduledJob struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job, including the schedule.
+    Spec ScheduledJobSpec
+
+    // Status is a structure describing current status of a job.
+    Status ScheduledJobStatus
+}
+
+// ScheduledJobList is a collection of scheduled jobs.
+type ScheduledJobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []ScheduledJob
+}
+```
+
+`ScheduledJobSpec` structure is defined to contain all the information how the actual
+job execution will look like, including the `JobSpec` from [Job Controller API](https://github.com/openshift/origin/blob/master/docs/proposals/job.md)
+and the schedule in ISO8601 format.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // Spec is a structure defining the expected behavior of a job.
+    Spec JobSpec
+
+    // Schedule contains the schedule in ISO8601 format.
+    Schedule string
+
+    // SkipOutdated specifies that if AllowConcurrent is false, only the newest job
+    // will be started (default: true), ignoring executions that missed their schedule.
+    SkipOutdated bool
```

Comment:
> Why do we need this? I would like to see something like:

Assume, that for some reason OpenShift was down, or at least the Schedule Job Controller missed a couple of executions, I don't want them to be rerun, just because they are outdated, I know that the next execution will deal with those outdated data still.

> So how about:
StartPolicy with values AllowConcurrent, CancelExisting ?

That makes sense and makes the api more readable.